pkg/storaged/dialog.js

```diff
@@ -72,16 +76,94 @@ define([
             $dialog.remove();
         });
 
-        $dialog.find('.selectpicker').selectpicker();
         $dialog.find('.dialog-arrow').on('click', toggle_arrow);
         $dialog.find('.dialog-select-row-table tbody').on('click', select_row);
         $dialog.find('.dialog-select-row-table tbody tr:first-child').addClass('highlight');
 
+        /* Size sliders
+         */
+
+        function setup_size_slider(field) {
+            var value = field.Value || field.Max;
+            var parent = $dialog.find('[data-field="' + field.SizeSlider + '"]');
+            var slider = controls.Slider();
+            var units = cockpit.get_byte_units(value);
+
+            parent.data('max', field.Max);
+            parent.data('round', field.Round);
+            parent.find('.slider').replaceWith(slider);
+            parent.find('.size-unit').append(units.map(function (u) {
+                return $('<option>', { value: u.factor, selected: u.selected }).text(u.name);
+            }));
+
+            $(slider).on('change', size_slider_changed);
+            parent.find('.size-text').on('change', size_text_changed);
+            parent.find('.size-unit').on('change', size_unit_changed);
+
+            slider.prop("value", value / field.Max);
+            slider.trigger("change", [value / field.Max]);
+        }
+
+        function size_slider_changed(event, value) {
+            var parent = $(this).parents('.size-slider');
+            var input = parent.find('.size-text');
+            var unit = parent.find('.size-unit');
+            var max = parent.data('max');
+            var round = parent.data('round');
+
+            value *= max;
+            if (round)
+                value = Math.round(value / round) * round;
+
+            parent.val(value);
+            input.val(cockpit.format_number(value / +unit.val()));
+        }
+
+        function size_text_changed(event) {
+            var input = $(this);
+            var parent = input.parents('.size-slider');
+            var unit = parent.find('.size-unit');
+            var unit_val = +unit.val();
+            var slider = parent.find('.slider');
+            var max = parent.data('max');
+            var value = +input.val() * unit_val;
+
+            // As a special case, if the user types something that
```

Comment:
How about adding "Bytes" as a unit?  That way, the user can enter a precise and unambiguous number.